src/test/java/hudson/remoting/PrefetchingTest.java

```diff
@@ -131,9 +136,9 @@ public void testGetResourceAsStream() throws Exception {
      * Validates that the resource is coming from a jar.
      */
     private void verifyResource(String v) throws IOException, InterruptedException {
-        assertTrue(v, v.startsWith("jar:file:"));
-        assertTrue(v, v.contains(dir.getPath()));
-        assertTrue(v, v.endsWith("::hello"));
+        Assert.assertThat(v, allOf(startsWith("jar:file:"), 
+                                   containsString(dir.toURI().getPath()), 
```

Comment:
Specifically `file.toUri()` is wrong; the only correct call (in Java 7+) is `file.toPath().toUri()`.